mesh/mesh.cpp

```diff
@@ -10761,7 +10761,7 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
 
    int i, j, k, l, s;
 
-   int nv;
+   int nv, nbe;
```

Comment:
I think it's better practice to declare the variables in the narrowest scope in which they are needed. Of course, this is a more invasive change, so it's a judgement call about whether it's worth it.

```diff
diff --git a/mesh/mesh.cpp b/mesh/mesh.cpp
index 298600861..bf1cc6dc4 100644
--- a/mesh/mesh.cpp
+++ b/mesh/mesh.cpp
@@ -10759,21 +10759,16 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
    MFEM_ASSERT(Dim == spaceDim, "2D Manifolds not supported\n");
    if (Dim != 3 && Dim != 2) { return; }
 
-   int i, j, k, l, s;
-
-   int nv, nbe;
-   const int *ind;
-
    int *vcount = new int[NumOfVertices];
-   for (i = 0; i < NumOfVertices; i++)
+   for (int i = 0; i < NumOfVertices; i++)
    {
       vcount[i] = 0;
    }
-   for (i = 0; i < NumOfElements; i++)
+   for (int i = 0; i < NumOfElements; i++)
    {
-      nv = elements[i]->GetNVertices();
-      ind = elements[i]->GetVertices();
-      for (j = 0; j < nv; j++)
+      int nv = elements[i]->GetNVertices();
+      const int *ind = elements[i]->GetVertices();
+      for (int j = 0; j < nv; j++)
       {
          vcount[ind[j]]++;
       }
@@ -10781,13 +10776,13 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
 
    int *voff = new int[NumOfVertices+1];
    voff[0] = 0;
-   for (i = 1; i <= NumOfVertices; i++)
+   for (int i = 1; i <= NumOfVertices; i++)
    {
       voff[i] = vcount[i-1] + voff[i-1];
    }
 
    int **vown = new int*[NumOfVertices];
-   for (i = 0; i < NumOfVertices; i++)
+   for (int i = 0; i < NumOfVertices; i++)
    {
       vown[i] = new int[vcount[i]];
    }
@@ -10799,30 +10794,30 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
       Transpose(ElementToEdgeTable(), edge_el);
 
       // Fake printing of the elements.
-      for (i = 0; i < NumOfElements; i++)
+      for (int i = 0; i < NumOfElements; i++)
       {
-         nv  = elements[i]->GetNVertices();
-         ind = elements[i]->GetVertices();
-         for (j = 0; j < nv; j++)
+         int nv  = elements[i]->GetNVertices();
+         const int *ind = elements[i]->GetVertices();
+         for (int j = 0; j < nv; j++)
          {
             vcount[ind[j]]--;
             vown[ind[j]][vcount[ind[j]]] = i;
          }
       }
 
-      for (i = 0; i < NumOfVertices; i++)
+      for (int i = 0; i < NumOfVertices; i++)
       {
          vcount[i] = voff[i+1] - voff[i];
       }
 
-      nbe = 0;
-      for (i = 0; i < edge_el.Size(); i++)
+      int nbe = 0;
+      for (int i = 0; i < edge_el.Size(); i++)
       {
          const int *el = edge_el.GetRow(i);
          if (edge_el.RowSize(i) > 1)
          {
-            k = partitioning[el[0]];
-            l = partitioning[el[1]];
+            int k = partitioning[el[0]];
+            int l = partitioning[el[1]];
             if (interior_faces || k != l)
             {
                nbe += 2;
@@ -10837,28 +10832,28 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
       // Print the type of the mesh and the boundary elements.
       out << "areamesh2\n\n" << nbe << '\n';
 
-      for (i = 0; i < edge_el.Size(); i++)
+      for (int i = 0; i < edge_el.Size(); i++)
       {
          const int *el = edge_el.GetRow(i);
          if (edge_el.RowSize(i) > 1)
          {
-            k = partitioning[el[0]];
-            l = partitioning[el[1]];
+            int k = partitioning[el[0]];
+            int l = partitioning[el[1]];
             if (interior_faces || k != l)
             {
                Array<int> ev;
                GetEdgeVertices(i,ev);
                out << k+1; // attribute
-               for (j = 0; j < 2; j++)
-                  for (s = 0; s < vcount[ev[j]]; s++)
+               for (int j = 0; j < 2; j++)
+                  for (int s = 0; s < vcount[ev[j]]; s++)
                      if (vown[ev[j]][s] == el[0])
                      {
                         out << ' ' << voff[ev[j]]+s+1;
                      }
                out << '\n';
                out << l+1; // attribute
-               for (j = 1; j >= 0; j--)
-                  for (s = 0; s < vcount[ev[j]]; s++)
+               for (int j = 1; j >= 0; j--)
+                  for (int s = 0; s < vcount[ev[j]]; s++)
                      if (vown[ev[j]][s] == el[1])
                      {
                         out << ' ' << voff[ev[j]]+s+1;
@@ -10868,12 +10863,12 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          }
          else
          {
-            k = partitioning[el[0]];
+            int k = partitioning[el[0]];
             Array<int> ev;
             GetEdgeVertices(i,ev);
             out << k+1; // attribute
-            for (j = 0; j < 2; j++)
-               for (s = 0; s < vcount[ev[j]]; s++)
+            for (int j = 0; j < 2; j++)
+               for (int s = 0; s < vcount[ev[j]]; s++)
                   if (vown[ev[j]][s] == el[0])
                   {
                      out << ' ' << voff[ev[j]]+s+1;
@@ -10884,13 +10879,13 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
 
       // Print the elements.
       out << NumOfElements << '\n';
-      for (i = 0; i < NumOfElements; i++)
+      for (int i = 0; i < NumOfElements; i++)
       {
-         nv  = elements[i]->GetNVertices();
-         ind = elements[i]->GetVertices();
+         int nv  = elements[i]->GetNVertices();
+         const int *ind = elements[i]->GetVertices();
          out << partitioning[i]+1 << ' '; // use subdomain number as attribute
          out << nv << ' ';
-         for (j = 0; j < nv; j++)
+         for (int j = 0; j < nv; j++)
          {
             out << ' ' << voff[ind[j]]+vcount[ind[j]]--;
             vown[ind[j]][vcount[ind[j]]] = i;
@@ -10898,17 +10893,17 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          out << '\n';
       }
 
-      for (i = 0; i < NumOfVertices; i++)
+      for (int i = 0; i < NumOfVertices; i++)
       {
          vcount[i] = voff[i+1] - voff[i];
       }
 
       // Print the vertices.
       out << voff[NumOfVertices] << '\n';
-      for (i = 0; i < NumOfVertices; i++)
-         for (k = 0; k < vcount[i]; k++)
+      for (int i = 0; i < NumOfVertices; i++)
+         for (int k = 0; k < vcount[i]; k++)
          {
-            for (j = 0; j < Dim; j++)
+            for (int j = 0; j < Dim; j++)
             {
                out << vertices[i](j) << ' ';
             }
@@ -10921,10 +10916,10 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
       out << "NETGEN_Neutral_Format\n";
       // print the vertices
       out << voff[NumOfVertices] << '\n';
-      for (i = 0; i < NumOfVertices; i++)
-         for (k = 0; k < vcount[i]; k++)
+      for (int i = 0; i < NumOfVertices; i++)
+         for (int k = 0; k < vcount[i]; k++)
          {
-            for (j = 0; j < Dim; j++)
+            for (int j = 0; j < Dim; j++)
             {
                out << ' ' << vertices[i](j);
             }
@@ -10933,12 +10928,12 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
 
       // print the elements
       out << NumOfElements << '\n';
-      for (i = 0; i < NumOfElements; i++)
+      for (int i = 0; i < NumOfElements; i++)
       {
-         nv = elements[i]->GetNVertices();
-         ind = elements[i]->GetVertices();
+         int nv = elements[i]->GetNVertices();
+         const int *ind = elements[i]->GetVertices();
          out << partitioning[i]+1; // use subdomain number as attribute
-         for (j = 0; j < nv; j++)
+         for (int j = 0; j < nv; j++)
          {
             out << ' ' << voff[ind[j]]+vcount[ind[j]]--;
             vown[ind[j]][vcount[ind[j]]] = i;
@@ -10946,17 +10941,19 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          out << '\n';
       }
 
-      for (i = 0; i < NumOfVertices; i++)
+      for (int i = 0; i < NumOfVertices; i++)
       {
          vcount[i] = voff[i+1] - voff[i];
       }
 
       // print the boundary information.
-      nbe = 0;
-      for (i = 0; i < NumOfFaces; i++)
-         if ((l = faces_info[i].Elem2No) >= 0)
+      int nbe = 0;
+      for (int i = 0; i < NumOfFaces; i++)
+      {
+         int l = faces_info[i].Elem2No;
+         if (l >= 0)
          {
-            k = partitioning[faces_info[i].Elem1No];
+            int k = partitioning[faces_info[i].Elem1No];
             l = partitioning[l];
             if (interior_faces || k != l)
             {
@@ -10967,28 +10964,31 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          {
             nbe++;
          }
+      }
 
       out << nbe << '\n';
-      for (i = 0; i < NumOfFaces; i++)
-         if ((l = faces_info[i].Elem2No) >= 0)
+      for (int i = 0; i < NumOfFaces; i++)
+      {
+         int l = faces_info[i].Elem2No;
+         if (l >= 0)
          {
-            k = partitioning[faces_info[i].Elem1No];
+            int k = partitioning[faces_info[i].Elem1No];
             l = partitioning[l];
             if (interior_faces || k != l)
             {
-               nv = faces[i]->GetNVertices();
-               ind = faces[i]->GetVertices();
+               int nv = faces[i]->GetNVertices();
+               int *ind = faces[i]->GetVertices();
                out << k+1; // attribute
-               for (j = 0; j < nv; j++)
-                  for (s = 0; s < vcount[ind[j]]; s++)
+               for (int j = 0; j < nv; j++)
+                  for (int s = 0; s < vcount[ind[j]]; s++)
                      if (vown[ind[j]][s] == faces_info[i].Elem1No)
                      {
                         out << ' ' << voff[ind[j]]+s+1;
                      }
                out << '\n';
                out << l+1; // attribute
-               for (j = nv-1; j >= 0; j--)
-                  for (s = 0; s < vcount[ind[j]]; s++)
+               for (int j = nv-1; j >= 0; j--)
+                  for (int s = 0; s < vcount[ind[j]]; s++)
                      if (vown[ind[j]][s] == faces_info[i].Elem2No)
                      {
                         out << ' ' << voff[ind[j]]+s+1;
@@ -10998,28 +10998,31 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          }
          else
          {
-            k = partitioning[faces_info[i].Elem1No];
-            nv = faces[i]->GetNVertices();
-            ind = faces[i]->GetVertices();
+            int k = partitioning[faces_info[i].Elem1No];
+            int nv = faces[i]->GetNVertices();
+            const int *ind = faces[i]->GetVertices();
             out << k+1; // attribute
-            for (j = 0; j < nv; j++)
-               for (s = 0; s < vcount[ind[j]]; s++)
+            for (int j = 0; j < nv; j++)
+               for (int s = 0; s < vcount[ind[j]]; s++)
                   if (vown[ind[j]][s] == faces_info[i].Elem1No)
                   {
                      out << ' ' << voff[ind[j]]+s+1;
                   }
             out << '\n';
          }
+      }
    }
    //  Dim is 3
    else if (meshgen == 2) // TrueGrid
    {
       // count the number of the boundary elements.
-      nbe = 0;
-      for (i = 0; i < NumOfFaces; i++)
-         if ((l = faces_info[i].Elem2No) >= 0)
+      int nbe = 0;
+      for (int i = 0; i < NumOfFaces; i++)
+      {
+         int l = faces_info[i].Elem2No;
+         if (l >= 0)
          {
-            k = partitioning[faces_info[i].Elem1No];
+            int k = partitioning[faces_info[i].Elem1No];
             l = partitioning[l];
             if (interior_faces || k != l)
             {
@@ -11030,7 +11033,7 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          {
             nbe++;
          }
-
+      }
 
       out << "TrueGrid\n"
           << "1 " << voff[NumOfVertices] << " " << NumOfElements
@@ -11040,17 +11043,17 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
           << "0.0 0.0 0.0 0 0 0.0 0.0 0 0.0\n"
           << "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n";
 
-      for (i = 0; i < NumOfVertices; i++)
-         for (k = 0; k < vcount[i]; k++)
+      for (int i = 0; i < NumOfVertices; i++)
+         for (int k = 0; k < vcount[i]; k++)
             out << voff[i]+k << " 0.0 " << vertices[i](0) << ' '
                 << vertices[i](1) << ' ' << vertices[i](2) << " 0.0\n";
 
-      for (i = 0; i < NumOfElements; i++)
+      for (int i = 0; i < NumOfElements; i++)
       {
-         nv = elements[i]->GetNVertices();
-         ind = elements[i]->GetVertices();
+         int nv = elements[i]->GetNVertices();
+         const int *ind = elements[i]->GetVertices();
          out << i+1 << ' ' << partitioning[i]+1; // partitioning as attribute
-         for (j = 0; j < nv; j++)
+         for (int j = 0; j < nv; j++)
          {
             out << ' ' << voff[ind[j]]+vcount[ind[j]]--;
             vown[ind[j]][vcount[ind[j]]] = i;
@@ -11058,32 +11061,34 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          out << '\n';
       }
 
-      for (i = 0; i < NumOfVertices; i++)
+      for (int i = 0; i < NumOfVertices; i++)
       {
          vcount[i] = voff[i+1] - voff[i];
       }
 
       // boundary elements
-      for (i = 0; i < NumOfFaces; i++)
-         if ((l = faces_info[i].Elem2No) >= 0)
+      for (int i = 0; i < NumOfFaces; i++)
+      {
+         int l = faces_info[i].Elem2No;
+         if (l >= 0)
          {
-            k = partitioning[faces_info[i].Elem1No];
+            int k = partitioning[faces_info[i].Elem1No];
             l = partitioning[l];
             if (interior_faces || k != l)
             {
-               nv = faces[i]->GetNVertices();
-               ind = faces[i]->GetVertices();
+               int nv = faces[i]->GetNVertices();
+               const int *ind = faces[i]->GetVertices();
                out << k+1; // attribute
-               for (j = 0; j < nv; j++)
-                  for (s = 0; s < vcount[ind[j]]; s++)
+               for (int j = 0; j < nv; j++)
+                  for (int s = 0; s < vcount[ind[j]]; s++)
                      if (vown[ind[j]][s] == faces_info[i].Elem1No)
                      {
                         out << ' ' << voff[ind[j]]+s+1;
                      }
                out << " 1.0 1.0 1.0 1.0\n";
                out << l+1; // attribute
-               for (j = nv-1; j >= 0; j--)
-                  for (s = 0; s < vcount[ind[j]]; s++)
+               for (int j = nv-1; j >= 0; j--)
+                  for (int s = 0; s < vcount[ind[j]]; s++)
                      if (vown[ind[j]][s] == faces_info[i].Elem2No)
                      {
                         out << ' ' << voff[ind[j]]+s+1;
@@ -11093,23 +11098,24 @@ void Mesh::PrintElementsWithPartitioning(int *partitioning,
          }
          else
          {
-            k = partitioning[faces_info[i].Elem1No];
-            nv = faces[i]->GetNVertices();
-            ind = faces[i]->GetVertices();
+            int k = partitioning[faces_info[i].Elem1No];
+            int nv = faces[i]->GetNVertices();
+            const int *ind = faces[i]->GetVertices();
             out << k+1; // attribute
-            for (j = 0; j < nv; j++)
-               for (s = 0; s < vcount[ind[j]]; s++)
+            for (int j = 0; j < nv; j++)
+               for (int s = 0; s < vcount[ind[j]]; s++)
                   if (vown[ind[j]][s] == faces_info[i].Elem1No)
                   {
                      out << ' ' << voff[ind[j]]+s+1;
                   }
             out << " 1.0 1.0 1.0 1.0\n";
          }
+      }
    }
 
    out << flush;
 
-   for (i = 0; i < NumOfVertices; i++)
+   for (int i = 0; i < NumOfVertices; i++)
    {
       delete [] vown[i];
    }
```

Reply:
I agree completely